src/redux/greet/greetSlice.js

```diff
@@ -0,0 +1,42 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const getGreetingsURL = 'http://localhost:3000/api/v1/greets';
+
+const initialState = {
+  loading: false,
+  greets: [],
+  error: '',
+};
+
+export const fetchGreets = createAsyncThunk('greet/fetchGreets', async () => {
+  const { data } = await (axios.get(getGreetingsURL));
+  return data;
+});
+
+/* eslint-disable no-param-reassign */
+
+const greetSlice = createSlice({
+  name: 'greet',
+  initialState,
+  extraReducers: (builder) => {
+    builder.addCase(fetchGreets.pending, (state) => {
+      state.loading = true;
+    })
+      .addCase(fetchGreets.fulfilled, (state, action) => {
+        state.loading = false;
+        state.error = '';
+        state.greets = action.payload;
+      })
+      .addCase(fetchGreets.rejected, (state) => {
+        state.loading = false;
+        state.error = '';
+      });
+  },
+});
+
```

Comment:
- [ ] Kindly remove the `eslint-disable` statements. `Eslint` helps us to write clean, readable and industry-standard JavaScript codes. Using `eslint-disable` statements defeats this purpose. Kindly remove it. 👍 😎 

#### Hint: For the _**no-param-reassign**_ error, you can read this part of the [documentation](https://redux-toolkit.js.org/usage/immer-reducers#:~:text=So%20if%20we%20can%27t%20change%20the%20originals%2C%20how%20do%20we%20return%20an%20updated%20state%3F) to properly reassign the `state`. 👍 😎